runtime/gc_glue_java/MetronomeDelegate.cpp

```diff
@@ -1653,7 +1660,11 @@ MM_MetronomeDelegate::scanContinuationObject(MM_EnvironmentRealtime *env, J9Obje
 
 		GC_VMThreadStackSlotIterator::scanSlots(currentThread, objectPtr, (void *)&localData, stackSlotIteratorForRealtimeGC, bStackFrameClassWalkNeeded, false);
 	}
-	return scanMixedObject(env, objectPtr);
+	if (MUTATOR_THREAD == env->getThreadType()) {
+		return 0;
+	} else {
+		return scanMixedObject(env, objectPtr);
+	}
```

Comment:
Let's do some cleanup. We've been calling this special method do/scanContinuationObject for various collectors, but the method was really only scanning java stack. At some point you wanted to introduce a wrapper to make it more clean, but  I think that just this whole method (family of methods) should be renamed to a relatively generic name like scanContinuationNativeSlots (or OffheapSlots). We could be more specific like too, scanContinuationJavaStack - up for discussion. 

Specifically for RT, that method should not call scanMixedObject at all (like for other GCs, it does not create a scanner). Then, 
- for preMount, what is in mutator thread context, we can call it directly, without any other action and context related checks (if mutator or GC thread).
- for object scan, in GC thread context, when we recognize this scan type, we call this method, and right after that we call scanMixedObject, and we are still free of any context related checks.

During

Reply:
Optionally, we could even have scanContinuationObject too, beside scanContinuationNativeSlots, that will be just a wrapper around scanContinuationNativeSlots and after that also call scanMixedObject or construct a scanner. That is to keep the code more uniform with some other specialized objects that need extra action (like scanReferenceObject)